### PR TITLE
Reworking Initial Project Setup

### PR DESCRIPTION
**Pull request by LightGuard**
Reworking how the initial Project Setup works. It references the 15 minute getting started sections for further information. It is very basic now, as depending on what the user wants to do, they will need to follow a different setup after the initial project creation.

### CODE REVIEW

**Review by jomarko — COMMENTED** [review]
@LightGuard hi, thank you for the PR, here is my feedback

### Creating a BAMOE Project Manually
Good. I like we link two template `pom.xml` files.

In the attached screenshot I marked them as (1) and (2). My comments are these. For the (1) I was not able to get rid of the error [1] until I removed/commented out [2] from the pom.xml. 

For (1) and (2) is probably good to mention a dmn (1) or bpmn (2) file needs to be in the project resources, otherwise similar error [3] appears running `mvn clean package`

Screenshot 2024-06-19 142055

### Create a new project using the BAMOE Canvas accelerator functionality.
We do not mention this on `project-setup.html` more, is there plan to add it, is that tracked in a ticket? or such procedure is already documented elsewhere? in such scenario we should probably add a link?

[1]
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.8.4:build (default) on project app: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.kie.kogito.event.EventEmitter and qualifiers [@Default]
[ERROR]         - injection target: org.kie.kogito.eventdriven.rules.QuarkusEventDrivenRulesController#eventEmitter
[ERROR]         - declared on CLASS bean [types=[org.kie.kogito.eventdriven.rules.QuarkusEventDrivenRulesController, java.lang.Object, org.kie.kogito.eventdriven.rules.EventDrivenRulesController], qualifiers=[@Default, @Any], target=org.kie.kogito.eventdriven.rules.QuarkusEventDrivenRulesController]
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1508)
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:320)
[ERROR]         at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:160)
[ERROR]         at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:488)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:842)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.kie.kogito.event.EventEmitter and qualifiers [@Default]
[ERROR]         - injection target: org.kie.kogito.eventdriven.rules.QuarkusEventDrivenRulesController#eventEmitter
[ERROR]         - declared on CLASS bean [types=[org.kie.kogito.eventdriven.rules.QuarkusEventDrivenRulesController, java.lang.Object, org.kie.kogito.eventdriven.rules.EventDrivenRulesController], qualifiers=[@Default, @Any], target=org.kie.kogito.eventdriven.rules.QuarkusEventDrivenRulesController]
[ERROR]         at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:518)
[ERROR]         at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:638)
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:308)
[ERROR]         ... 13 more
[ERROR] -> [Help 1]
```

[2]
```
        <dependency>
            <groupId>org.drools</groupId>
            <artifactId>drools-addons-quarkus-events-rules</artifactId>
        </dependency>
```

[3]
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.8.4:build (default) on project app: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.kie.kogito.process.Processes and qualifiers [@Default]
[ERROR]         - injection target: org.kie.kogito.jobs.embedded.EmbeddedJobExecutor#processes
[ERROR]         - declared on CLASS bean [types=[org.kie.kogito.jobs.service.executor.JobExecutor, java.lang.Object, org.kie.kogito.jobs.embedded.EmbeddedJobExecutor], qualifiers=[@Default, @Any], target=org.kie.kogito.jobs.embedded.EmbeddedJobExecutor]
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1508)
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:320)
[ERROR]         at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:160)
[ERROR]         at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:488)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:842)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.kie.kogito.process.Processes and qualifiers [@Default]
[ERROR]         - injection target: org.kie.kogito.jobs.embedded.EmbeddedJobExecutor#processes
[ERROR]         - declared on CLASS bean [types=[org.kie.kogito.jobs.service.executor.JobExecutor, java.lang.Object, org.kie.kogito.jobs.embedded.EmbeddedJobExecutor], qualifiers=[@Default, @Any], target=org.kie.kogito.jobs.embedded.EmbeddedJobExecutor]
[ERROR]         at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:518)
[ERROR]         at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:638)
[ERROR]         at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:308)
[ERROR]         ... 13 more
[ERROR] -> [Help 1]
```

File: getting-started/project-setup.adoc [hunk 1]
```diff
@@ -26,60 +26,26 @@ xref:../partials/rules-sample-pom.adoc[Sample Rules/Decision Application POM Tem
...
+Follow along with one of the examples in 15 minutes to complete setup and usage for the various use cases: xref:business-rule-microservice.html[business rules microservice], xref:orchestrating.html[orchestration], or xref:decision-microservice.html[decision microservice].
 
 == Source Control
-Canvas only works with git.
+{CANVAS}, a standalone editor for BAMOE files, only works with git.
 Any git based tools should continue to work with BAMOE projects (i.e. vscode, git cli, GitHub Desktop, etc.).
-Add the code to git using a preferred tool, and push to a remote.
\ No newline at end of file
+Add the code to git using a preferred tool, and push to a remote.
+More information about installing and using {CANVAS} can be found later in this documentation.
```
Comment (justlmills) [inline getting-started/project-setup.adoc:51]: Add a link to ../installation/canvas.adoc

**Review by justlmills — APPROVED** [review]

File: getting-started/project-setup.adoc [hunk 2]
```diff
@@ -26,60 +26,26 @@ xref:../partials/rules-sample-pom.adoc[Sample Rules/Decision Application POM Tem
...
-. Click the `pricing.dmn` file to open it. 
-  When you do so, you may be greeted with a message similar to `The current DMN does not have any Diagram associated with it. Do you want to auto-generate it?` 
-  If so click `Auto-generate Diagram` to automap the DMN locations.
-+
-image::automatic-layout.png[Automatic Layout]
-
-. When the diagram opens you will see something similar to below, and you can start exploring it. 
-  The DMN consists of two inputs: *Age* and *Previous incidents?*, which are used to make the decision, *Base price*.
-+
-image::dmn-first-view.png[DMN First view]
```
Comment (porcelli) [inline getting-started/project-setup.adoc:59]: feels like we are loosing too much

File: getting-started/project-setup.adoc [hunk 3]
```diff
@@ -26,60 +26,26 @@ xref:../partials/rules-sample-pom.adoc[Sample Rules/Decision Application POM Tem
...
-+
-image::open-decision.png[Edit DMN Decision]
-
-. From here you will see the Decision Table that is associated with the Base Price decision. 
-  From here you will see two (*2*) input columns (`Age` and `Previous Incidents`), as well as one output column (`Base price`) all with their types below them. 
-  These types are controlled from the properties panel similarly to how they were opened when we previously looked at `Age`. 
-  This decision has 4 different rows that could fire, with a Hit Policy of `UNIQUE` signified by the **U** in the top left corner of the table. 
-  A decision writer could make any comments they want to the table and have them saved with the decision here.
-+
-image::dt-stable.png[Viewing DMN Decision]
```
Comment (porcelli) [inline getting-started/project-setup.adoc:80]: I think we are loosing content here

**Review by porcelli — CHANGES_REQUESTED** [review]

File: getting-started/project-setup.adoc [hunk 4]
```diff
@@ -26,60 +26,26 @@ xref:../partials/rules-sample-pom.adoc[Sample Rules/Decision Application POM Tem
 
 xref:../partials/process-sample-pom.adoc[Sample Process Application POM Template] a sample POM that describes a typical process project dependencies and plugins.
 
-Build the project in the root folder:
-[source,console]
-----
-mvn clean package
-----
-
```
[the remark this answers is not in the document]
Comment (LightGuard) [reply getting-started/project-setup.adoc:34]: We're using this as a starting and then letting people move on to one of the other "15 minute" sections.

File: getting-started/project-setup.adoc (hunk 2 above)
Comment (LightGuard) [reply getting-started/project-setup.adoc:59]: I think we overlap in other sections

File: getting-started/project-setup.adoc (hunk 3 above)
Comment (LightGuard) [reply getting-started/project-setup.adoc:80]: I think this is the same comment as above

**Pull request merged**